employee-rostering-frontend/src/util/ImmutableCollectionOperations.ts

```diff
@@ -96,3 +97,39 @@ export function mapWithUpdatedElement<T extends DomainObject>(map: Map<number, D
   return mapWithElement(map, updatedElement);
 }
 
+// An immutable version of a collection with a lot of helpful methods
+export class Stream<T> {
```

Comment:
Time to switch to one of the libraries out there?
- http://staltz.github.io/xstream/
- https://rxjs.dev/guide/operators
- https://immutable-js.github.io/immutable-js/

Reply:
+1 to immutable-js; the other libraries solves different use cases (side note: might also invalidate the entire file ImmutableCollectionOperations).

Reply:
OK, I'll file a JIRA for that. Let's not add that to this PR.

Reply:
https://issues.redhat.com/browse/PLANNER-1752